src/videojs-hls.js

```diff
@@ -1035,7 +1035,7 @@ videojs.Hls.translateMediaIndex = function(mediaIndex, original, update) {
   // bitrate switches should be happening here.
   translatedMediaIndex = (mediaIndex + (original.mediaSequence - update.mediaSequence));
 
-  if (translatedMediaIndex >= update.segments.length || translatedMediaIndex < 0) {
+  if (translatedMediaIndex > update.segments.length || translatedMediaIndex < 0) {
```

Comment:
When all the segments currently available in a live playlist have been downloaded, mediaIndex will point to one beyond the end. There's nothing wrong with this situation, we're just waiting for a playlist refresh. It's incorrect to reset mediaIndex back to the live point if this occurs.